### PR TITLE
add product maturing cucumbers

### PR DESCRIPTION
**Pull request by adi-stefan**

### CODE REVIEW

File: backend/de.metas.cucumber/src/test/java/de/metas/cucumber/stepdefs/pporder/PP_Order_Candidate_StepDef.java [hunk 1]
```diff
@@ -668,4 +702,17 @@ private String getPPOrderCandidateContextForProductId(@NonNull final ProductId p
 
 		return "Current PP_Order_Candidates available for M_Product_ID:\n\n" + messageBuilder;
 	}
+
+	@And("after not more than {int}s, no PP_Order_Candidates are found for Issue_HU_ID: {string}")
+	public void validatePP_Order_Candidate(
+			final int timeoutSec,
+			final String issueHuIdIdentifier) throws InterruptedException
```
Comment (metas-rc) [inline backend/de.metas.cucumber/src/test/java/de/metas/cucumber/stepdefs/pporder/PP_Order_Candidate_StepDef.java:709]: "IdIdentifier" sounds like a pleonasm. I suggest you name this variable "issueHUIdentifier"

File: backend/de.metas.cucumber/src/test/java/de/metas/cucumber/stepdefs/pporder/PP_Order_StepDef.java [hunk 2]
```diff
@@ -527,4 +538,28 @@ private static Map<String, ArrayList<String>> getPPOrderCandIdentifier2PPOrderId
 
 		return ppOrderCandidate2PPOrderIdsOrdered;
 	}
+
+	@And("load manufactured HU for PP_Order:")
+	public void loadManufacturedHUs(@NonNull final DataTable dataTable)
+	{
+		DataTableRow.toRows(dataTable)
+				.forEach(this::loadMAnufacturedHu);
```
Comment (metas-rc) [inline backend/de.metas.cucumber/src/test/java/de/metas/cucumber/stepdefs/pporder/PP_Order_StepDef.java:546]: Typo "MAnufactured"

File: backend/de.metas.cucumber/src/test/java/de/metas/cucumber/stepdefs/pporder/PP_Order_StepDef.java [hunk 3]
```diff
@@ -527,4 +538,28 @@ private static Map<String, ArrayList<String>> getPPOrderCandIdentifier2PPOrderId
...
+	{
+		final IQueryBuilder<I_PP_Order_Qty> builder = queryBL.createQueryBuilder(I_PP_Order_Qty.class)
+				.addEqualsFilter(I_PP_Order_Qty.COLUMNNAME_PP_Order_BOMLine_ID, null);
+		final I_PP_Order ppOrder = tableRow.getAsOptionalIdentifier(I_PP_Order_Qty.COLUMNNAME_PP_Order_ID)
+				.map(ppOrderTable::get)
+				.orElseThrow(() -> new AdempiereException("No PP_Order found for identifier " + tableRow.getAsString(I_PP_Order_Qty.COLUMNNAME_PP_Order_ID)));
+		builder.addEqualsFilter(I_PP_Order_Qty.COLUMNNAME_PP_Order_ID, ppOrder.getPP_Order_ID())
+				.addEqualsFilter(I_PP_Order_Qty.COLUMNNAME_M_Product_ID, ppOrder.getM_Product_ID());
+
+		final I_M_HU manufacturedHu = builder.andCollect(I_PP_Order_Qty.COLUMN_M_HU_ID)
```
Comment (metas-rc) [inline backend/de.metas.cucumber/src/test/java/de/metas/cucumber/stepdefs/pporder/PP_Order_StepDef.java:559]: Can't ``I_PP_Order_Qty.COLUMNNAME_M_HU_ID`` be used as parameter?

File: backend/de.metas.cucumber/src/test/java/de/metas/cucumber/stepdefs/pporder/maturing/M_Maturing_Configuration_StepDef.java [hunk 4]
```diff
@@ -0,0 +1,122 @@
...
+import static org.adempiere.model.InterfaceWrapperHelper.saveRecord;
+
+@FieldDefaults(level = AccessLevel.PRIVATE, makeFinal = true)
+public class M_Maturing_Configuration_StepDef
+{
+	M_Maturing_Configuration_StepDefData maturingConfigurationTable;
+	M_Maturing_Configuration_Line_StepDefData maturingConfigurationLineTable;
+	M_Product_StepDefData productStepDefData;
+	IQueryBL queryBL = Services.get(IQueryBL.class);
+
```
Comment (metas-rc) [inline backend/de.metas.cucumber/src/test/java/de/metas/cucumber/stepdefs/pporder/maturing/M_Maturing_Configuration_StepDef.java:49]: all of the above: final

File: backend/de.metas.cucumber/src/test/java/de/metas/cucumber/stepdefs/pporder/PP_Order_Candidate_StepDef.java (hunk 1 above)
Comment (adi-stefan) [reply backend/de.metas.cucumber/src/test/java/de/metas/cucumber/stepdefs/pporder/PP_Order_Candidate_StepDef.java:709]: done.

File: backend/de.metas.cucumber/src/test/java/de/metas/cucumber/stepdefs/pporder/maturing/M_Maturing_Configuration_StepDef.java (hunk 4 above)
Comment (adi-stefan) [reply backend/de.metas.cucumber/src/test/java/de/metas/cucumber/stepdefs/pporder/maturing/M_Maturing_Configuration_StepDef.java:49]: they already are:
Line42: `@FieldDefaults(level = AccessLevel.PRIVATE, makeFinal = true)`

File: backend/de.metas.cucumber/src/test/java/de/metas/cucumber/stepdefs/pporder/PP_Order_StepDef.java (hunk 2 above)
Comment (adi-stefan) [reply backend/de.metas.cucumber/src/test/java/de/metas/cucumber/stepdefs/pporder/PP_Order_StepDef.java]: done.

File: backend/de.metas.cucumber/src/test/java/de/metas/cucumber/stepdefs/pporder/PP_Order_StepDef.java (hunk 3 above)
Comment (adi-stefan) [reply backend/de.metas.cucumber/src/test/java/de/metas/cucumber/stepdefs/pporder/PP_Order_StepDef.java:559]: as discussed, both options could be used, none are deprecated. If it's an informal best practice, let's formalize it by deprecating one API and patch managing that PR to all active branches.

**Review by metas-rc — APPROVED** [review]

**Pull request merged**